trl/trainer/sft_trainer.py

```diff
@@ -145,6 +145,8 @@ def __init__(
             output_dir = "tmp_trainer"
             warnings.warn(f"No `SFTConfig` passed, using `output_dir={output_dir}`.")
             args = SFTConfig(output_dir=output_dir)
+        elif args is not None and args.__class__.__name__ == "TrainingArguments":
+            args = SFTConfig(**args.to_dict())
```

Comment:
Nice fix!

Reply:
This still fails for me with error: `TypeError: SFTConfig.__init__() got an unexpected keyword argument 'cache_dir'` because TrainingArguments has a field for `cache_dir` but SFTConfig does not

Reply:
Hmm that's strange, there is no field `cache_dir` in TrainingArguments: https://github.com/huggingface/transformers/blob/main/src/transformers/training_args.py - can you show us how do you get that error?

Reply:
Apologies I see this is mixture of TrainingArguments where my project built a class off of transformers.TrainingArguments that we pass into trl.SFTTrainer and so the additional parameters that we are adding to TrainingArguments is causing the discrepancy issue.